src/runtime/internal/ResizeObserverSingleton.ts

```diff
@@ -0,0 +1,60 @@
+const MapImplementation = "WeakMap" in window ? WeakMap : Map;
+
+/**
+ * Resize observer singleton
+ * https://groups.google.com/a/chromium.org/g/blink-dev/c/z6ienONUb5A/m/F5-VcUZtBAAJ
+ */
+export class ResizeObserverSingleton {
```

Comment:
It's a little unfortunate that this much code is needed to have a performant implementation of the resize observer. This adds quite a bit of code to the runtime (only if you use the feature of course), but I guess that's unavoidable..

Reply:
If we can guarantee one listener per singleton, it can be simplified considerably.

I think we can??

Reply:
What do you mean by "one listener per singleton"?

Reply:
I created 3 singleton instances for each box size. clipRect and contentBox share the same one.

I had to do this because they use different observe options.

If we use 4 observers total, we might be able to guarantee 1 listener per observer for each element.

Reply:
yes you could guarantee that then, because it's an error to have multiple `bind:X` with the same `X` on one element.

Reply:
Fixed

Reply:
Turns out we only need 3. Just need to combine the event names.